dj_rql/drf/backend.py

```diff
@@ -59,14 +82,25 @@ def get_filter_class(view):
     def get_query(cls, filter_instance, request, view):
         return get_query(request)
 
-    @staticmethod
-    def _get_filter_instance(filter_class, queryset, view):
-        qual_name = '{0}.{1}'.format(view.basename, filter_class.__name__)
+    @classmethod
+    def _get_or_init_cache(cls, filter_class, view):
+        qual_name = cls._get_filter_cls_qual_name(filter_class, view)
+        return cls._CACHES.setdefault(
+            qual_name, filter_class.QUERIES_CACHE_BACKEND(int(filter_class.QUERIES_CACHE_SIZE)),
+        )
 
-        filter_instance = FilterCache.CACHE.get(qual_name)
+    @classmethod
+    def _get_filter_instance(cls, filter_class, queryset, view):
+        qual_name = cls._get_filter_cls_qual_name(filter_class, view)
+
+        filter_instance = _FilterClassCache.CACHE.get(qual_name)
         if filter_instance:
             return filter_class(queryset=queryset, instance=filter_instance)
 
         filter_instance = filter_class(queryset)
-        FilterCache.CACHE[qual_name] = filter_instance
+        _FilterClassCache.CACHE[qual_name] = filter_instance
         return filter_instance
+
+    @staticmethod
+    def _get_filter_cls_qual_name(filter_class, view):
+        return '{0}.{1}'.format(view.basename, filter_class.__name__)
```

Comment:
There is collision possible because view.basename and filter_class.__name__ have no unique constraints.

Suggest to add some unique view identificator:

`str(id(view))`

`'{0}.{1}.{2}'.format(view.basename, filter_class.__name__, id(view))`

`'{0}.{1}'.format(view.__class__.__module__, view.__class__.__name__)`